### PR TITLE
Find VS Code installations with Spotlight query on macOS

### PR DESCRIPTION
**Pull request by jmagman**
In addition to hard-coded `Applications` VS Code install locations, on macOS also query [Spotlight](https://support.apple.com/en-gb/HT204014) (the app search engine when you cmd-space) for all installs by searching with the command line tool `mdfind` for the app's bundle identifier (`com.microsoft.VSCode` or `com.microsoft.VSCodeInsiders`).

See also #80475 and #80479

### CODE REVIEW

File: packages/flutter_tools/lib/src/vscode/vscode.dart [hunk 1]
```diff
@@ -110,15 +115,26 @@ class VsCode {
   // macOS Extensions:
   //   $HOME/.vscode/extensions
   //   $HOME/.vscode-insiders/extensions
-  static List<VsCode> _installedMacOS(FileSystem fileSystem, Platform platform) {
+  static List<VsCode> _installedMacOS(FileSystem fileSystem, Platform platform, ProcessManager processManager) {
     final String? homeDirPath = FileSystemUtils(fileSystem: fileSystem, platform: platform).homeDirPath;
-    return _findInstalled(<_VsCodeInstallLocation>[
-      _VsCodeInstallLocation(
+    // Query Spotlight for unexpected installation locations.
+    final ProcessResult vsCodeSpotlightQueryResult = processManager.runSync(<String>[
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/vscode/vscode.dart:121]: do we catch process exceptions in a higher scope? operating under the assumption this binary will either be removed or renamed some day.

Comment (jmagman) [reply packages/flutter_tools/lib/src/vscode/vscode.dart:121]: Good point.  Let me add that.

Comment (jmagman) [reply packages/flutter_tools/lib/src/vscode/vscode.dart]: Done.

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**